libs/utils/analysis/latency_analysis.py

```diff
@@ -391,14 +377,87 @@ def plotLatency(self, task, kind='all', tag=None, threshold_ms=1, bins=64):
 
         # Join the two data frames
         df = wkp_df.append(prt_df)
-        ymax = 1.1 * df.latency.max()
+        cdf = self._getCDF(df.latency, (threshold_ms / 1000.))
+
+        return df, cdf
+
+
+    @memoized
+    def _dfg_latency_stats_df(self, task, kind='all', threshold_ms=1):
+        """
+        Compute statistics on latencies of the specified task.
+
+        :param task: the task to report latencies for
+        :type task: int or list(str)
+
+        :param kind: the kind of latencies to report (WAKEUP and/or PREEMPT")
+        :type kind: str
+
+        :param threshold_ms: the minimum acceptable [ms] value to report
+                             graphically in the generated plots
+        :type threshold_ms: int or float
+
+        :returns: a DataFrame with statistics on task latencies
+        """
+
+        # Get latency events
+        df, cdf = self._get_latency_df(task, kind, threshold_ms)
         self._log.info('Total: %5d latency events', len(df))
```

Comment:
Do we want to have all of this logging spilled out when we just want to use the dataframe ? Maybe that should only be logged in the plotting version ?

Reply:
Mmm... good point!